engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala

```diff
@@ -141,29 +141,42 @@ class EngineJobExecutionActor(replyTo: ActorRef,
   when(Pending) {
     case Event(Execute, NoData) =>
       increment(NonEmptyList("jobs", List("ejea", "executing", "starting")))
-      requestExecutionToken()
-      goto(RequestingExecutionToken)
+      if (restarting) {
+        requestRestartCheckToken()
+        goto(RequestingRestartCheckToken)
+      } else {
+        replyTo ! JobStarting(jobDescriptorKey)
+        requestExecutionToken()
+        goto(RequestingExecutionToken)
+      }
   }
 
-  when(RequestingExecutionToken) {
+  // This condition only applies for restarts
+  when(RequestingRestartCheckToken) {
     case Event(JobTokenDispensed, NoData) =>
       replyTo ! JobStarting(jobDescriptorKey)
-      if (restarting) {
-        val jobStoreKey = jobDescriptorKey.toJobStoreKey(workflowIdForLogging)
-        jobStoreActor ! QueryJobCompletion(jobStoreKey, jobDescriptorKey.call.outputPorts.toSeq)
-        goto(CheckingJobStore)
-      } else {
-        requestValueStore()
-      }
+      val jobStoreKey = jobDescriptorKey.toJobStoreKey(workflowIdForLogging)
+      jobStoreActor ! QueryJobCompletion(jobStoreKey, jobDescriptorKey.call.outputPorts.toSeq)
+      goto(CheckingJobStore)
+  }
+
+  when(RequestingExecutionToken) {
+    case Event(JobTokenDispensed, NoData) =>
+      if (!restarting)
+        replyTo ! JobStarting(jobDescriptorKey)
```

Comment:
Do we send this twice for non-restarters (here and on line 146)?

Reply:
oops yes, good catch.